dev/tasks/linux-packages/yum/arrow.spec.in

```diff
@@ -200,10 +272,80 @@ Libraries and header files for Apache Arrow Datasets.
 %files dataset-devel
 %defattr(-,root,root,-)
 %doc README.md LICENSE.txt NOTICE.txt
+%{_includedir}/arrow/dataset/
 %{_libdir}/libarrow_dataset.a
 %{_libdir}/libarrow_dataset.so
 %{_libdir}/pkgconfig/arrow-dataset.pc
 
+%if %{use_flight}
+%package flight-libs
+Summary:	C++ library for fast data transport.
+License:	Apache-2.0
+Requires:	%{name}-libs = %{version}-%{release}
+Requires:	%{name}-python-libs = %{version}-%{release}
+Requires:	openssl
+
+%description flight-libs
+This package contains the libraries for Apache Arrow Flight.
+
+%files flight-libs
+%defattr(-,root,root,-)
+%doc README.md LICENSE.txt NOTICE.txt
+%{_libdir}/libarrow_flight.so.*
+%{_libdir}/libarrow_python_flight.so.*
+
+%package flight-devel
+Summary:	Libraries and header files for Apache Arrow Flight.
+License:	Apache-2.0
+Requires:	%{name}-flight-libs = %{version}-%{release}
+Requires:	%{name}-python-devel = %{version}-%{release}
+
+%description flight-devel
+Libraries and header files for Apache Arrow Flight.
+
+%files flight-devel
+%defattr(-,root,root,-)
+%doc README.md LICENSE.txt NOTICE.txt
+%{_includedir}/arrow/flight/
+%{_libdir}/libarrow_flight.a
+%{_libdir}/libarrow_flight.so
+%{_libdir}/libarrow_python_flight.a
+%{_libdir}/libarrow_python_flight.so
+%{_libdir}/pkgconfig/arrow-flight.pc
+%endif
+
+%if %{use_gandiva}
+%package -n gandiva-libs
+Summary:	C++ library for compiling and evaluating exressions on Apache Arrow data.
+License:	Apache-2.0
+Requires:	%{name}-libs = %{version}-%{release}
+Requires:	ncurses-libs
+
+%description -n gandiva-libs
+This package contains the libraries for Gandiva.
+
+%files -n gandiva-libs
+%defattr(-,root,root,-)
+%doc README.md LICENSE.txt NOTICE.txt
+%{_libdir}/libgandiva.so.*
+
+%package -n gandiva-devel
+Summary:	Libraries and header files for Gandiva.
+License:	Apache-2.0
+Requires:	gandiva-libs = %{version}-%{release}
```

Comment:
I don't know if it requries llvm-devel. There's no mention which file is meant to be public and/or private.

```
find cpp/src/gandiva/ -type f -name '*.h' | xargs grep include|grep llvm
cpp/src/gandiva/llvm_includes.h:#include <llvm/ExecutionEngine/ExecutionEngine.h>
cpp/src/gandiva/llvm_includes.h:#include <llvm/IR/IRBuilder.h>
cpp/src/gandiva/llvm_includes.h:#include <llvm/IR/LLVMContext.h>
cpp/src/gandiva/llvm_includes.h:#include <llvm/IR/Module.h>
cpp/src/gandiva/expr_validator.h:#include "gandiva/llvm_types.h"
cpp/src/gandiva/lvalue.h:#include "gandiva/llvm_includes.h"
cpp/src/gandiva/llvm_types.h:#include <map>
cpp/src/gandiva/llvm_types.h:#include <vector>
cpp/src/gandiva/llvm_types.h:#include "gandiva/arrow.h"
cpp/src/gandiva/llvm_types.h:#include "gandiva/llvm_includes.h"
cpp/src/gandiva/llvm_types.h:#include "gandiva/logging.h"
cpp/src/gandiva/llvm_types.h:#include "gandiva/visibility.h"
cpp/src/gandiva/llvm_generator.h:#include <cstdint>
cpp/src/gandiva/llvm_generator.h:#include <memory>
cpp/src/gandiva/llvm_generator.h:#include <string>
cpp/src/gandiva/llvm_generator.h:#include <vector>
cpp/src/gandiva/llvm_generator.h:#include "arrow/util/macros.h"
cpp/src/gandiva/llvm_generator.h:#include "gandiva/annotator.h"
cpp/src/gandiva/llvm_generator.h:#include "gandiva/compiled_expr.h"
cpp/src/gandiva/llvm_generator.h:#include "gandiva/configuration.h"
cpp/src/gandiva/llvm_generator.h:#include "gandiva/dex_visitor.h"
cpp/src/gandiva/llvm_generator.h:#include "gandiva/engine.h"
cpp/src/gandiva/llvm_generator.h:#include "gandiva/execution_context.h"
cpp/src/gandiva/llvm_generator.h:#include "gandiva/function_registry.h"
cpp/src/gandiva/llvm_generator.h:#include "gandiva/gandiva_aliases.h"
cpp/src/gandiva/llvm_generator.h:#include "gandiva/llvm_types.h"
cpp/src/gandiva/llvm_generator.h:#include "gandiva/lvalue.h"
cpp/src/gandiva/llvm_generator.h:#include "gandiva/selection_vector.h"
cpp/src/gandiva/llvm_generator.h:#include "gandiva/value_validity_pair.h"
cpp/src/gandiva/llvm_generator.h:#include "gandiva/visibility.h"
cpp/src/gandiva/function_ir_builder.h:#include "gandiva/llvm_types.h"
cpp/src/gandiva/engine.h:#include "gandiva/llvm_includes.h"
cpp/src/gandiva/engine.h:#include "gandiva/llvm_types.h"
cpp/src/gandiva/compiled_expr.h:#include "gandiva/llvm_includes.h"
```

Reply:
Ah, we need to depend on `llvm-devel`.
I missed it. Thanks.